src/raven.js

```diff
@@ -469,6 +469,30 @@ Raven.prototype = {
         return this;
     },
 
+    /*
+     * Adds a shouldSendCallback callback that chains after any previously set
+     * callback(s).
+     *
+     * @param {function} callback The callback to run which allows
+     *                            introspecting the blob before sending
+     * @return {Raven}
+     */
+    addShouldSendCallback: function(callback) {
+        var prevCallback = this._globalOptions.shouldSendCallback;
```

Comment:
Sigh, again hesitant to add more API methods – even though I know this solves some serious problems.

We've talked about just merging `shouldSendCallback` / `dataCallback` in a major version change (4.0).